xrtpy/response/xrt_teem.py

```diff
@@ -0,0 +1,735 @@
+            & ((T_error - T_e) <= np.log10(Te_err_threshold))
+            & (Kd1 <= photon_noise_threshold)
+            & (Kd2 <= photon_noise_threshold)
+        )
+        if verbose:
+            print(f"number of pixels ruled out by threshold = " f"{np.sum(~ok_pixel)}")
+            print(f"number of pixels ruled out by T_e errors = {np.sum(~tthr)}")
+            print(f"number of pixels ruled out by d1 noise  = {np.sum(~k1thr)}")
+            print(f"number of pixels ruled out by d2 noise  = {np.sum(~k2thr)}")
+            print(f"number of bad pixels before threshold   = " f"{np.sum(~ok_wothr)}")
+        mask = mask | ~ok_pixel
+        T_e[mask] = 0.0
+        EM[mask] = 0.0
+        T_error[mask] = 0.0
+        EMerror[mask] = 0.0
+
+        if verbose:
+            print("from xrt_teem:")
+            Tmodel = tresp1.CHIANTI_temperature.value
+            if Trange:
+                Tmodel = Tmodel[
+                    (np.log10(Tmodel) >= Trange[0]) & (np.log10(Tmodel) <= Trange[1])
+                ]
+            print(f"Examined T_e range: {Tmodel.min()} - {Tmodel.max()} K")
+            print(f"Applied thresholds: - T_e error < {Te_err_threshold*100.} %")
+            print(
+                f"                    - Photon noise < "
+                f"{photon_noise_threshold*100.} %"
+            )
+    else:
+        if verbose:
+            Tmodel = tresp1.CHIANTI_temperature.value
+            print("from xrt_teem:")
+            print(f"Examined T_e range: {Tmodel.min()} - {Tmodel.max()} K")
+            print("No thresholds applied")
+    Tmap, EMmap, Terrmap, EMerrmap = make_results_maps(
+        hdr1, hdr2, T_e, EM, T_error, EMerror, mask
+    )
+    return Tmap, EMmap, Terrmap, EMerrmap
+
+
+def rebin_image(data, binfac=1):
+    """
+    Given a data array and a binning factor return the data array rebinned by
+    the binning factor. Note: the size of the original data array is
+    preserved, despite the adding up of adjacent pixels. Thus the sum of all
+    the pixels will be increased by the factor binfac.
+    """
+
+    s = data.shape
+    ns = (s[0] // binfac, s[1] // binfac)
+    rbs = (ns[0], binfac, ns[1], binfac)
+    # sums the data in binfac x binfac sized regions
+    drbin = data.reshape(rbs).sum(-1).sum(1)
+    # for a boolean mask, this makes a pixel masked if any of the summed
+    # pixels is masked. If we want to mask only if all the pixels are masked
+    # then we could use prod in place of sum above
+
+    if data.dtype == bool:
+        # need to convert back to bool after summing
+        drbin = drbin.astype(bool)
+    # This restores the image to the size of the original images as in the
+    # IDL code:
+    if data.dtype == bool:
+        dtmp = np.zeros_like(data, dtype=bool)
+    else:
+        dtmp = np.zeros_like(data)
+    for i in range(binfac):
+        for j in range(binfac):
+            dtmp[i::binfac, j::binfac] = drbin[:]
+    data = dtmp
+    return data
+
+
+def deriv(x, y):
```

Comment:
```suggestion
def _deriv(x, y):
```
I'd be inclined for this to not be in our public API either.